pytorch_lightning/trainer/states.py

```diff
@@ -0,0 +1,31 @@
+from enum import Enum, auto
+
+from pytorch_lightning import Callback
+
+
+class TrainerState(Enum):
+    """ State which is set to the Trainer to indicate what is being executed. """
+    INITIALIZE = auto()
+    RUNNING = auto()
+    FINISHED = auto()
+
+
+class _TrainerStateSwitcher(Callback):
+    """ Special callback used by the Trainer. This callback sets proper
+        state to the trainer depending on what is being executed.
+    """
+
+    def on_init_start(self, trainer):
+        trainer.state = TrainerState.INITIALIZE
+
+    def on_init_end(self, trainer):
```

Comment:
not need as it is the same as on init start

Reply:
Done